FWCore/Framework/src/ProductResolvers.cc

```diff
@@ -168,12 +167,12 @@ namespace edm {
         // The file may already be closed so the reader is invalid
         return;
       }
-      if (mcc and (branchType == InEvent || branchType == InProcess) and aux_) {
+      if (mcc and branchType == InEvent and aux_) {
```

Comment:
If I am remembering correctly (it's been several months since I edited this file), this only affects the ActivityRegistry signal, preModuleDelayedGetSignal_ and the corresponding post signal. These require a StreamContext argument so that signal cannot be used. While ProcessBlock products are being read, the context is a GlobalContext.

Possibly I should add separate signals for ProcessBlock? Is that needed? It should be doing delayed get for ProcessBlock products, although this is happening during prefetching if I am remembering how all this works. It is not getting everything immediately as it does for Runs and LuminosityBlocks.

Reply:
I think there is also an issue here with aux_ not being set when analogous ProcessBlock signals would need to fire.

Reply:
Ah, right, these are only about ActivityRegistry signals. 

> Possibly I should add separate signals for ProcessBlock?

Good question, I don't know. For completeness the answer would be "yes", but I'm not sure if there would be any actual benefit (which would favor "no").